### PR TITLE
enh(unattended): manange supported Centreon versions

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

- Added possibility to select supported Centreon versions (21.10, 22.04, 22.10, 23.04, 23.10)
- Do not install PHP and MariaDB repo and binaries for poller type servers
- Added glibc langpack for ES/PT/FR/DE
- Added Debian OS support (central & poller mode)
- Added Raspberry support for poller mode only
- Added update mode
- Added update silent mode for central
- Added possibility to set admin password

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by tuntoja — APPROVED** [review]

File: centreon/unattended.sh [hunk 1]
```diff
@@ -252,17 +255,41 @@ function pause() {
...
+			;;
+		"Red Hat"*)
+			detected_os_release="redhat-release-${OS_VERSIONID}"
+			;;
+		Rocky*)
+			detected_os_release="rocky-release-${OS_VERSIONID}"
+			;;
+		*)
+			log "ERROR" "Unsupported distribution ${OS_NAME} detected"
+			error_and_exit "This '$script_short_name' script only supports Red Hat compatible distributions. Please check https://docs.centreon.com/docs/installation/introduction for alternative installation methods."
```
Comment (sc979) [inline centreon/unattended.sh:288]: Debian is not redhat compatible

File: centreon/unattended.sh [hunk 2]
```diff
@@ -610,11 +716,12 @@ function enable_new_services() {
 			log "DEBUG" "On central..."
 			systemctl enable mariadb $OS_SPEC_SERVICES snmpd snmptrapd gorgoned centreontrapd cbd centengine centreon
 			systemctl restart mariadb $OS_SPEC_SERVICES snmpd snmptrapd
+			systemctl start centreontrapd snmptrapd
```
Comment (paul-oureib) [inline centreon/unattended.sh:719]:
```suggestion
			systemctl start centreontrapd
```

File: centreon/unattended.sh [hunk 3]
```diff
@@ -733,21 +1172,33 @@ function update_after_installation() {
...
-	set_selinux_config $selinux_mode
+		set_selinux_config $selinux_mode
+	fi
 }
 #========= end of function update_after_installation()
 
+#========= begin of function test_password_policy()
+function test_password_policy() {
+	if [[ ${#centreon_admin_password} -ge 12 && "${centreon_admin_password}" == *[A-Z]* && "${centreon_admin_password}" == *[a-z]* && "${centreon_admin_password}" == *[0-9]* && "${centreon_admin_password}" == *[\!@#$%^\&*()\\[\]{}\-_+=~\`\|\:\;\"\'\<\>\,\.\/\?]* ]]; then
+        log "INFO" "Password is complient with Centreon security policy"
```
Comment (paul-oureib) [inline centreon/unattended.sh:1195]:
```suggestion
        log "INFO" "Password is compliant with Centreon security policy"
```

File: centreon/unattended.sh [hunk 4]
```diff
@@ -733,21 +1172,33 @@ function update_after_installation() {
...
+    else
+        error_and_exit "Password is not complient with Centreon security policy ([A-Z][a-z][0-9][\!@#$%^\&*()\\[\]{}\-_+=~\`\|\:\;\"\'\<\>\,\.\/\?]{12,})"
```
Comment (paul-oureib) [inline centreon/unattended.sh:1197]:
```suggestion
        error_and_exit "Password is not compliant with Centreon security policy ([A-Z][a-z][0-9][\!@#$%^\&*()\\[\]{}\-_+=~\`\|\:\;\"\'\<\>\,\.\/\?]{12,})"
```

**Review by sc979 — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Pull request merged**